src/input/input_handler.cpp

```diff
@@ -0,0 +1,694 @@
+                                                                      : 0.5f;
+            controller->SetTouchpadState(0, new_button_state, touchpad_x, 0.5f);
+            controller->CheckButton(0, button, new_button_state);
+            break;
+        case LEFTJOYSTICK_HALFMODE:
+            leftjoystick_halfmode = new_button_state;
+            break;
+        case RIGHTJOYSTICK_HALFMODE:
+            rightjoystick_halfmode = new_button_state;
+            break;
+        case KEY_TOGGLE:
+            if (new_button_state) {
+                // LOG_DEBUG(Input, "Toggling a button...");
+                ToggleKeyInList(new_param);
+            }
+            break;
+        default: // is a normal key (hopefully)
+            controller->CheckButton(0, (OrbisPadButtonDataOffset)button, new_button_state);
+            break;
+        }
+    } else if (axis != Axis::AxisMax) {
+        float multiplier = 1.0;
+        switch (axis) {
+        case Axis::LeftX:
+        case Axis::LeftY:
+            multiplier = leftjoystick_halfmode ? 0.5 : 1.0;
+            break;
+        case Axis::RightX:
+        case Axis::RightY:
+            multiplier = rightjoystick_halfmode ? 0.5 : 1.0;
+            break;
+        case Axis::TriggerLeft:
+        case Axis::TriggerRight:
+            controller->Axis(0, axis, GetAxis(0x0, 0x80, new_param));
+            // Also handle button counterpart for TriggerLeft and TriggerRight
+            controller->CheckButton(0,
+                                    axis == Axis::TriggerLeft ? OrbisPadButtonDataOffset::L2
+                                                              : OrbisPadButtonDataOffset::R2,
+                                    new_param > 0x20);
+            return;
+        default:
+            break;
+        }
+        controller->Axis(0, axis, GetAxis(-0x80, 0x80, new_param * multiplier));
+    }
+}
+
+// Updates the list of pressed keys with the given input.
+// Returns whether the list was updated or not.
+bool UpdatePressedKeys(u32 value, bool is_pressed) {
+    // Skip invalid inputs
+    if (value == (u32)-1) {
+        return false;
+    }
+    if ((value & 0x80000000) != 0) {
+        // analog input, it gets added when it first sends an event,
+        // and from there, it only changes the parameter
+        // reverse iterate until we get out of the 0x8000000 range, if found,
+        // update the parameter, if not, add it to the end
+        u32 value_to_search = value & 0xF00FFFFF;
+        for (auto& it = --pressed_keys.end(); (it->first & 0x80000000) != 0; it--) {
+            if ((it->first & 0xF00FFFFF) == value_to_search) {
+                it->first = value;
+                // LOG_DEBUG(Input, "New value for {:X}: {:x}", value, value);
+                return true;
+            }
+        }
+        // LOG_DEBUG(Input, "Input activated for the first time, adding it to the list");
+        pressed_keys.insert(pressed_keys.end(), {value, false});
```

Comment:
This inserts the key into pressed_keys without checking is_pressed, is it okay?
```suggestion
        pressed_keys.emplace_back(value, false);
```